src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -857,6 +809,7 @@ def _collect_metai_classpath(self, metacp_result, relative_input_paths):
     #    TODO remove this after https://github.com/scalameta/scalameta/issues/1791 is released
```

Comment:
Off topic, but: this is going to be problematic for #6679 I think.

Reply:
Yeah. Looks like I need to implement something for https://github.com/scalameta/scalameta/issues/1791 sooner rather than later--or fall back to a sort and search based approach.

Hm. Maybe a sort based approach would be more resilient. It bothers me that I've spent this much time on this particular awkward file munging problem.